src/evaluate.cpp

```diff
@@ -23,48 +23,122 @@
+                    // cerr << "&data = " << &data << endl;
+                    // cerr << "&gEmbededNNUEData = " << &gEmbededNNUEData << endl;
+                    // cerr << "&(gEmbededNNUEData[0]) = " << &(gEmbededNNUEData[0]) << endl;
+                    // cerr << "sizeof(data) = " << sizeof(data) << endl;
+                    // data = &(gEmbededNNUEData[0]);
+                    // std::memcpy(&data, &(gEmbededNNUEData[0]), sizeof(data));
+                    // std::memcpy(data, &gEmbededNNUEData, sizeof(data));
+                    
+                    char*  data   = const_cast<char*>(reinterpret_cast<const char*>(gEmbededNNUEData));
+                    size_t length = size_t(gEmbededNNUESize );
+                    
+                    cerr << "Trying to load eval from memory... " << eval_file << endl;
+                    if (   eval_file == EvalFileDefaultName
+                        && load_eval_from_memory(eval_file, data, length))
+                    {
+                        eval_file_loaded = eval_file;
+                        cerr << "Eval loaded from memory: OK" << endl;
+                    }
+                }
+                else
+                {
+                    cerr << "Trying to find eval file on disc... " << directory + eval_file << endl;
+                    if (load_eval_file(directory + eval_file))
+                    {
+                        eval_file_loaded = eval_file;
+                        cerr << "Eval loaded from disc: OK" << endl;
+                    }
+                }
+            }
   }
 
+  /// verify_NNUE() verifies that the last net used was loaded successfully
   void verify_NNUE() {
 
-    std::string eval_file = std::string(Options["EvalFile"]);
+    string eval_file = string(Options["EvalFile"]);
+
     if (useNNUE && eval_file_loaded != eval_file)
     {
         UCI::OptionsMap defaults;
         UCI::init(defaults);
 
-        sync_cout << "info string ERROR: NNUE evaluation used, but the network file " << eval_file << " was not loaded successfully." << sync_endl;
-        sync_cout << "info string ERROR: The UCI option EvalFile might need to specify the full path, including the directory/folder name, to the file." << sync_endl;
-        sync_cout << "info string ERROR: The default net can be downloaded from: https://tests.stockfishchess.org/api/nn/"+std::string(defaults["EvalFile"]) << sync_endl;
-        sync_cout << "info string ERROR: If the UCI option Use NNUE is set to true, network evaluation parameters compatible with the program must be available." << sync_endl;
-        sync_cout << "info string ERROR: The engine will be terminated now." << sync_endl;
-        std::exit(EXIT_FAILURE);
+        string msg1 = "If the UCI option Use NNUE is set to true, network evaluation parameters compatible with the engine must be available.";
+        string msg2 = "You asked for NNUE evaluation, but the network file " + eval_file + " was not loaded successfully.";
```

Comment:
I wouldn't use 'You' seems a bit direct.

Reply:
Done